src/libpmem2/map_windows.c

```diff
@@ -147,16 +146,42 @@ pmem2_map(const struct pmem2_config *cfg, const struct pmem2_source *src,
 	if (ret)
 		return ret;
 
-	/* create a file mapping handle */
+	DWORD proto = PAGE_READWRITE;
 	DWORD access = FILE_MAP_ALL_ACCESS;
-	HANDLE mh = create_mapping(src->handle, offset, length,
-			PAGE_READWRITE, &err);
-	if (err == ERROR_ACCESS_DENIED) {
-		mh = create_mapping(src->handle, offset, length,
-				PAGE_READONLY, &err);
+
+	/* translate protection flags into Windows flags */
+	/* on Windows, PROT_WRITE implies PROT_READ */
```

Comment:
then flags==PMEM2_PROT_WRITE should fail because otherwise, we are doing something other than what user requested